src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -1694,6 +1694,45 @@ MethodTableBuilder::BuildMethodTableThrowing(
         &pByValueClassCache, bmtMFDescs, bmtFP,
         &totalDeclaredFieldSize);
 
+    if (IsValueClass())
+    {
+        const void* pVal;                  // The custom value.
+        ULONG       cbVal;                 // Size of the custom value.
+        HRESULT hr = GetCustomAttribute(bmtInternal->pType->GetTypeDefToken(),
+            WellKnownAttribute::InlineArrayAttribute,
+            &pVal, &cbVal);
+
+        if (hr != S_FALSE)
+        {
+            if (bmtEnumFields->dwNumInstanceFields != 1)
+            {
+                // TODO: diagnostics, must have one field.
+                BuildMethodTableThrowException(IDS_CLASSLOAD_GENERAL);
+            }
+
+            if (cbVal >= (sizeof(INT32) + 2))
+            {
+                INT32 repeat = GET_UNALIGNED_VAL32((byte*)pVal + 2);
+                if (repeat > 0)
+                {
+                    bmtFP->NumInlineArrayElements = repeat;
+                    GetHalfBakedClass()->SetInlineArrayFlag();
+                }
+                else
+                {
+                    // TODO: diagnostics, repeat must be > 0
+                    BuildMethodTableThrowException(IDS_CLASSLOAD_GENERAL);
+                }
+
+                if (HasExplicitFieldOffsetLayout())
+                {
+                    // TODO: diagnostics, must not have explicit offsets
```

Comment:
Same here, do we need custom error? 
I am not sure what level of friendliness is expected from loader failures.